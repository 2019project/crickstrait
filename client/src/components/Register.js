import React, { Component } from "react";
import { createUser } from "../actions/Users";
import { connect } from "react-redux";
import "../styles/Register.css";
import { Link } from "react-router-dom";

export class Register extends Component {
  state = {
    name: "",
    email: "",
    phone: "",
    password: "",
    confirmPassword: "",
    showError: false,
    errors: {},
    isAdmin: false,
    login: false,
    signup: true
  };

  register = e => {
    e.preventDefault();
    if (this.validateForm()) {
      let user = {
        name: this.state.name,
        email: this.state.email,
        phone: this.state.phone,
        password: this.state.password,
        isAdmin: this.state.isAdmin
      };
      this.props.createUser(user);
      this.setState({
        name: "",
        email: "",
        phone: "",
        password: "",
        confirmPassword: ""
      });
    }
  };

  onChange = event => {
    this.setState({ [event.target.name]: event.target.value });
  };

  validateForm() {
    console.log("Hello");
    let errors = {};
    let formIsValid = true;

    if (!this.state.name) {
      formIsValid = false;
      errors["name"] = "*Please enter your name";
    }

    if (typeof this.state.name !== "undefined") {
      if (!this.state.name.match(/^[a-zA-Z ]*$/)) {
        formIsValid = false;
        errors["name"] = "*Please enter alphabet characters only.";
      }
    }

    if (!this.state.email) {
      formIsValid = false;
      errors["email"] = "*Please enter your email-ID.";
    }

    if (typeof this.state.email !== "undefined") {
      //regular expression for email validation
      var pattern = new RegExp(
        /^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i
      );
      if (!pattern.test(this.state.email)) {
        formIsValid = false;
        errors["email"] = "*Please enter valid email-ID.";
      }
    }

    if (!this.state.phone) {
      formIsValid = false;
      errors["phone"] = "*Please enter your mobile no.";
    }

    if (typeof this.state.phone !== "undefined") {
      if (!this.state.phone.match(/^[0-9]{10}$/)) {
        formIsValid = false;
        errors["phone"] = "*Please enter valid mobile no.";
      }
    }

    if (!this.state.password) {
      formIsValid = false;
      errors["password"] = "*Please enter your password.";
    }

    // if (typeof this.state.password !== "undefined") {
    //   if (
    //     !this.state.password.match(
    //       /^.*(?=.{8,})(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%&]).*$/
    //     )
    //   ) {
    //     formIsValid = false;
    //     errors["password"] = "*Please enter secure and strong password.";
    //   }
    // }

    if (!this.state.password) {
      formIsValid = false;
      errors["confirmPassword"] = "Please Enter Password Again";
    }

    if (this.state.password != this.state.confirmPassword) {
      formIsValid = false;
      errors["confirmPassword"] = "*Password Don't Match";
    }

    this.setState({
      errors: errors
    });
    return formIsValid;
  }

  render() {
    return (
      <div>
        <div id="registerMainContainer">
          <div id="registerLeftContainer">
            <h1 id="welcome" className="w3-container w3-center w3-animate-left">
              Welcome ,
            </h1>
            <p className="w3-container w3-center w3-animate-left">
              An Investment in Knowledge <br />
              pays the best Interest
            </p>
          </div>
          <div id="registerRightContainer">
            <div>
              <Link to="/login">
                <button
                  id={this.state.login ? "loginButton-active" : "loginButton"}
                >
                  Login
                </button>
              </Link>
              <button
                id={
                  this.state.signup ? "registerButton-active" : "registerButton"
                }
                onClick={() => {
                  this.setState({
                    login: false,
                    signup: true
                  });
                }}
              >
                Sign Up
              </button>
            </div>

            <form id="registerForm">
              <p>
                <input
                  id="registerInput"
                  placeholder="Name"
                  type="text"
                  name="name"
                  onChange={this.onChange}
                  value={this.state.name}
                />
              </p>
              <p>
                <p id="registerErrorMessage">{this.state.errors.name}</p>
              </p>
              <p>
                <input
                  id="registerInput"
                  placeholder="Email"
                  type="text"
                  name="email"
                  onChange={this.onChange}
                  value={this.state.email}
                />
              </p>
              <p id="registerErrorMessage">{this.state.errors.email}</p>
              <p>
                <input
                  id="registerInput"
                  placeholder="Mobile No"
                  type="text"
                  name="phone"
                  onChange={this.onChange}
                  value={this.state.phone}
                />
              </p>
              <p id="registerErrorMessage">{this.state.errors.phone}</p>
              <p>
                <input
                  id="registerInput"
                  placeholder="Password"
                  type="password"
                  name="password"
                  onChange={this.onChange}
                  value={this.state.password}
                />
              </p>
              <p id="registerErrorMessage">{this.state.errors.password}</p>
              <p>
                <input
                  id="registerInput"
                  placeholder="Confirm Password"
                  type="password"
                  name="confirmPassword"
                  onChange={this.onChange}
                  value={this.state.confirmPassword}
                />
              </p>
              <p id="registerErrorMessage">
                {this.state.errors.confirmPassword}
              </p>

              {this.props.error ? (
                <p id="registerErrorMessage">{this.props.error}</p>
              ) : (
                <p style={{ display: "none" }}>Hello</p>
              )}
              <p>
                <button onClick={this.register} id="registerFormButton">
                  Sign Up
                </button>
              </p>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  users: state.usersReducer.users,
  error: state.usersReducer.error
});

export default connect(
  mapStateToProps,
  { createUser }
)(Register);
