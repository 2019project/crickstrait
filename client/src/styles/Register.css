@import url(https://fonts.googleapis.com/css?family=Karla);
@import url("https://www.w3schools.com/w3css/4/w3.css");
* {
  font-family: Karla;
}

#registerMainContainer {
  width: 100%;
  grid-template-columns: "side1";
  background-color: #3e5871;
  display: grid;
  text-align: center;
  user-select: none;
}

#registerLeftContainer {
  color: #fff;
  font-size: 20px;
}

#welcome {
  font-size: 40px;
}

#registerRightContainer {
  color: #fff;
  padding: 20px;
}

#registerInput {
  border: none;
  outline: none;
  background: transparent;
  color: #fff;
  border-bottom: 1px solid #cacaca;
  padding: 18px;
  width: 70%;
  font-size: 16px;
}

#registerInput::placeholder {
  color: #b0b0b0;
}

#registerInput:focus,
#registerInput:active {
  transform: scaleX(1.1);
  border-bottom: 2px solid #39abf7;
}

#registerFormButton {
  padding: 15px;
  background: transparent;
  outline: none;
  font-size: 16px;
  border: 3px solid #39abf7;
  color: #39abf7;
  margin-top: 20px;
  cursor: pointer;
}

#registerErrorMessage {
  color: #ff4d4d;
  font-size: 14px;
}

#loginButton {
  background: transparent;
  color: #cacaca;
  outline: none;
  padding: 20px;
  font-weight: 500;
  font-size: 20px;
  border: none;
  cursor: pointer;
}

#loginButton-active {
  background: transparent;
  color: #fff;
  outline: none;
  padding: 20px;
  font-size: 20px;
  border: none;
  border-bottom: 2px solid #39abf7;
}

#registerButton-active {
  background: transparent;
  color: #fff;
  outline: none;
  padding: 20px;
  font-size: 20px;
  border: none;
  border-bottom: 2px solid #39abf7;
}

#registerButton {
  background: transparent;
  color: #cacaca;
  outline: none;
  padding: 20px;
  font-weight: 500;
  font-size: 20px;
  cursor: pointer;
  border: none;
}

#registerResetPassword {
  margin-left: 310px;
  color: #39abf7;
}

#registerFormButton:hover {
  color: #fff;
  right: 0;
  background-color: #39abf7;
  box-shadow: 0 12px 16px 0 rgba(231, 226, 226, 0.24),
    0 17px 50px 0 rgba(231, 226, 226, 0.24);
}

@media screen and (min-width: 430px) {
  #registerMainContainer {
    grid-template-rows: 675px;
    grid-template-columns: repeat(2, 1fr);
  }
  #registerMainContainer > div:nth-child(2) > form input {
    width: 50%;
  }

  #registerMainContainer > div:nth-child(2) {
    margin: auto 0px;
  }

  #registerMainContainer > div:nth-child(1) {
    margin: 150px auto;
  }

  #registerLeftContainer > h1 {
    font-size: 60px;
  }

  #registerLeftContainer > p {
    text-align: left;
    font-size: 30px;
  }
}
