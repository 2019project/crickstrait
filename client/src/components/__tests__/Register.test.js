import React from "react";
import { shallow, mount } from "enzyme";
import { Register } from "../Register";

const register = jest.fn();
const createUser = jest.fn();
const preventDefault = jest.fn();
const wrapper = shallow(
  <Register
    register={register}
    createUser={createUser}
    preventDefault={preventDefault}
  />
);

describe("Testing Register Component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });
  it("should mount the component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should have a container with name registerMainContainer", () => {
    expect(wrapper.find("#registerMainContainer")).toBeTruthy();
  });

  it("should check for links", () => {
    expect(wrapper.find("#loginButton").text()).toBe("Login");
  });

  it("should check if  Link component is present or not", () => {
    expect(wrapper.find("Link").exists()).toBeTruthy();
  });

  it("should have a container with name registerLeftContainer", () => {
    expect(wrapper.find("#registerLeftContainer")).toBeTruthy();
  });

  it("should have a container with name registerRightContainer", () => {
    expect(wrapper.find("#registerRightContainer")).toBeTruthy();
  });

  it("should have a form with name registerForm", () => {
    expect(wrapper.find("#registerForm")).toBeTruthy();
  });

  it("should simulate button click on Login button tab ", () => {
    wrapper.find("#registerButton-active").simulate("click");
    expect(wrapper.state().login).toBe(false);
    expect(wrapper.state().signup).toBe(true);
  });

  it("should check for Name input ", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(0)
        .props().placeholder
    ).toBe("Name");
  });

  it("should check for Email input ", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(1)
        .props().placeholder
    ).toBe("Email");
  });

  it("should check for Mobile No input ", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(2)
        .props().placeholder
    ).toBe("Mobile No");
  });

  it("should check for Password input ", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(3)
        .props().placeholder
    ).toBe("Password");
  });

  it("should check for Confirm Password input ", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(4)
        .props().placeholder
    ).toBe("Confirm Password");
  });

  it("should check for presense of register function", () => {
    let register = jest.spyOn(wrapper.instance(), "register");
    expect(register).toBeTruthy();
  });

  it("should check for presense of onChange function", () => {
    let onChange = jest.spyOn(wrapper.instance(), "onChange");
    expect(onChange).toBeTruthy();
  });

  it("shoudl check for input value", () => {
    const e = {
      target: {
        name: "name",
        value: "Bhavan"
      }
    };
    expect(wrapper.state().search_term).toBe(undefined);
    wrapper.instance().onChange(e);
    expect(createUser).toBeCalledTimes(0);
  });

  it("shoudl check for input value for email", () => {
    const e = {
      target: {
        name: "email",
        value: "testcase@gmail.com"
      }
    };
    expect(wrapper.state().search_term).toBe(undefined);
    wrapper.instance().onChange(e);
    expect(createUser).toBeCalledTimes(0);
  });

  it("there should be type in input fields,search and in select options", () => {
    expect(
      wrapper
        .find("#registerInput")
        .at(0)
        .prop("type")
    ).toBe("text");
    expect(
      wrapper
        .find("#registerInput")
        .at(1)
        .prop("type")
    ).toBe("text");
    expect(
      wrapper
        .find("#registerInput")
        .at(2)
        .prop("type")
    ).toBe("text");
    expect(
      wrapper
        .find("#registerInput")
        .at(3)
        .prop("type")
    ).toBe("password");
    expect(
      wrapper
        .find("#registerInput")
        .at(4)
        .prop("type")
    ).toBe("password");
  });

  it("checks for on edit button function to be called", () => {
    const e = { preventDefault: () => {} };
    jest.spyOn(e, "preventDefault");
    wrapper.find("#registerFormButton").simulate("click", e);
    expect(e.preventDefault).toBeCalled();
  });

  it("checks for Sign Up button click ", () => {
    wrapper.find("#registerFormButton").simulate("click");
  });

  it("testing for name input field", () => {
    const event = {
      target: {
        name: "name",
        value: "Bhavan"
      }
    };
    expect(wrapper.state().name).toBe(event.target.value);
  });
});
