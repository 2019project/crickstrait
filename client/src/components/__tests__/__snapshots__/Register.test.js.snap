// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Register Component should mount the component 1`] = `
<div>
  <div
    id="registerMainContainer"
  >
    <div
      id="registerLeftContainer"
    >
      <h1
        className="w3-container w3-center w3-animate-left"
        id="welcome"
      >
        Benefits of SignUp:
      </h1>
      <p
        className="w3-container w3-center w3-animate-left"
      >
        <ul
          id="registerBenefits"
        >
          <li>
            Detailed Comparison between Companies
          </li>
          <li>
            Can Download Financial Reports
          </li>
        </ul>
      </p>
    </div>
    <div
      id="registerRightContainer"
    >
      <div>
        <Link
          to="/login"
        >
          <button
            id="loginButton"
          >
            Login
          </button>
        </Link>
        <button
          id="registerButton-active"
          onClick={[Function]}
        >
          Sign Up
        </button>
      </div>
      <form
        id="registerForm"
      >
        <p>
          <input
            id="registerInput"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value=""
          />
        </p>
        <p>
          <p
            id="registerErrorMessage"
          />
        </p>
        <p>
          <input
            id="registerInput"
            name="email"
            onChange={[Function]}
            placeholder="Email"
            type="text"
            value=""
          />
        </p>
        <p
          id="registerErrorMessage"
        />
        <p>
          <input
            id="registerInput"
            name="phone"
            onChange={[Function]}
            placeholder="Mobile No"
            type="text"
            value=""
          />
        </p>
        <p
          id="registerErrorMessage"
        />
        <p>
          <input
            id="registerInput"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </p>
        <p
          id="registerErrorMessage"
        />
        <p>
          <input
            id="registerInput"
            name="confirmPassword"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
          />
        </p>
        <p
          id="registerErrorMessage"
        />
        <p
          style={
            Object {
              "display": "none",
            }
          }
        >
          Hello
        </p>
        <p>
          <button
            id="#registerFormButton"
            onClick={[Function]}
          >
            Proceed
          </button>
        </p>
      </form>
    </div>
  </div>
</div>
`;
